### PR TITLE
[Mailer] Add more information about sending email async

### PR DESCRIPTION
**Pull request by fabpot**
The example only works as of 6.2 thanks to https://github.com/symfony/symfony/pull/47075

I've written this PR because we keep having issues like https://github.com/symfony/symfony/issues/44439.
So, documenting a bit more how it works internally might help people understand what to do.

### CODE REVIEW

File: mailer.rst [hunk 1]
```diff
@@ -1204,9 +1204,40 @@ you have a transport called ``async``, you can route the message there:
...
 
+Thanks to this, instead of being delivered immediately, messages will be sent
+to the transport to be handled later (see :ref:`messenger-worker`). Note that
+the "rendering" of the email (computed headers, body rendering, ...) is also
+deferred and will only happen just before the email is sent by the Messenger
+handler.
 
-Thanks to this, instead of being delivered immediately, messages will be sent to
-the transport to be handled later (see :ref:`messenger-worker`).
+.. versionadded:: 6.2
```
Comment (OskarStark) [inline mailer.rst:1213]: Imho this should be a warning, otherwise it gets removed in Symfony 7 docs

File: mailer.rst [hunk 2]
```diff
@@ -1204,9 +1204,40 @@ you have a transport called ``async``, you can route the message there:
...
+
+When sending an email asynchronously, its instance must be serializable.
+This is always the case for :class:`Symfony\\Bridge\\Twig\\Mime\\Email`
+instances, but when sending a
+:class:`Symfony\\Bridge\\Twig\\Mime\\TemplatedEmail`, you must ensure that
+the ``context`` is serializable. If you have non-serializable variables,
+like Doctrine entities, either replace them with more specific variables or
+render the email before calling ``$mailer->send($email)``:
+
+.. code-block:: php
```
Comment (OskarStark) [inline mailer.rst]:
```suggestion
render the email before calling ``$mailer->send($email)``::
```

File: mailer.rst (hunk 1 above)
Comment (fabpot) [reply mailer.rst:1213]: Having this removed in 7.0 is a good thing.

File: mailer.rst (hunk 2 above)
Comment (fabpot) [reply mailer.rst]: fixed

**Comment by fabpot** [comment]
Related PR has been merged now.

File: mailer.rst (hunk 1 above)
Comment (OskarStark) [reply mailer.rst:1213]: Done then

**Review by OskarStark — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thank you Fabien.

**Pull request merged**